runtime/src/main/java/com/redhat/ipaas/rtdb/SQLRealTimeDB.java

```diff
@@ -0,0 +1,287 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.rtdb;
+
+import com.redhat.ipaas.rest.EventBus;
+import com.redhat.ipaas.rest.v1.controller.handler.rtdb.GetOptions;
+import com.redhat.ipaas.rest.v1.controller.handler.rtdb.RTDBException;
+import com.redhat.ipaas.rest.v1.controller.handler.rtdb.RealTimeDB;
+import org.keycloak.common.util.Base64;
+import org.skife.jdbi.v2.*;
+import org.skife.jdbi.v2.tweak.ResultSetMapper;
+import org.skife.jdbi.v2.util.IntegerColumnMapper;
+
+import javax.ws.rs.core.StreamingOutput;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.Random;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.redhat.ipaas.rtdb.Strings.*;
+
+/**
+ * Implements the RealTimeDB via DBI/JDBC
+ *
+ * Each value in the JSON tree is stored a simple record with the primary key being the
+ * path to the value.
+ */
+public class SQLRealTimeDB implements RealTimeDB {
+
+    private final DBI dbi;
+    private final EventBus bus;
+
+    // These values are used to compute a seq key
+    private long lastTimestamp = System.currentTimeMillis();
+    private final byte randomnessByte;
+    private long randomnessLong;
+
+    public SQLRealTimeDB( DBI dbi, EventBus bus) {
+        this.dbi = dbi;
+        this.bus = bus;
+
+        Random random = new Random();
+        randomnessByte = (byte) random.nextInt();
+        randomnessLong = random.nextLong();
+    }
+
+    public void createTables() {
+        withTransaction(dbi -> {
+            dbi.update("CREATE TABLE rtdb (path VARCHAR PRIMARY KEY, value VARCHAR, kind INT)");
+        });
+    }
+
+    public void dropTables() {
+        withTransaction(dbi -> {
+            dbi.update("DROP TABLE rtdb");
+        });
+    }
+
+
+    @Override
+    public String createKey() {
```

Comment:
Chances are low, within the same millisecond 2 nodes would need to have generated the same random number.